packages/operation/src/evm/utils.ts

```diff
@@ -6,29 +6,22 @@ export function deriveEndpointId(oisTitle: string, endpointName: string): string
 }
 
 export function deriveExtendedPublicKey(mnemonic: string): string {
-  const wallet = ethers.Wallet.fromMnemonic(mnemonic);
-  const hdNode = ethers.utils.HDNode.fromMnemonic(wallet.mnemonic.phrase);
+  const hdNode = ethers.utils.HDNode.fromMnemonic(mnemonic);
   return hdNode.neuter().extendedKey;
 }
 
-export function deriveAirnodeId(masterWalletAddress: string): string {
-  return ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['address'], [masterWalletAddress]));
+export function deriveWalletFromPath(mnemonic: string, provider: ethers.providers.JsonRpcProvider, path?: string) {
+  return ethers.Wallet.fromMnemonic(mnemonic, path).connect(provider);
 }
 
-export function deriveWalletFromPath(mnemonic: string, path: string, provider: ethers.providers.JsonRpcProvider) {
-  const masterHdNode = ethers.utils.HDNode.fromMnemonic(mnemonic);
-  const designatorHdNode = masterHdNode.derivePath(path);
-  return new ethers.Wallet(designatorHdNode.privateKey, provider);
-}
-
-export function getDesignatedWallet(mnemonic: string, requester: string, provider: ethers.providers.JsonRpcProvider) {
-  return deriveWalletFromPath(mnemonic, `m/0/${addressToDerivationPath(requester)}`, provider);
+export function getDesignatedWallet(mnemonic: string, provider: ethers.providers.JsonRpcProvider, sponsor: string) {
```

Comment:
Maybe a bit clearer?

```suggestion
export function getDesignatedWallet(mnemonic: string, provider: ethers.providers.JsonRpcProvider, sponsorAddress: string) {
```